cmake/kokkos_tribits.cmake

```diff
@@ -248,18 +248,18 @@ function(KOKKOS_SET_LIBRARY_PROPERTIES LIBRARY_NAME)
 
   list(APPEND ALL_KOKKOS_COMPILER_FLAGS ${KOKKOS_COMPILE_OPTIONS})
   target_compile_options(
-    ${LIBRARY_NAME} PUBLIC $<$<COMPILE_LANGUAGE:${KOKKOS_COMPILE_LANGUAGE}>:${KOKKOS_COMPILE_OPTIONS}>
+    ${LIBRARY_NAME} PUBLIC $<$<COMPILE_LANGUAGE:${KOKKOS_COMPILE_LANGUAGE},CXX>:${KOKKOS_COMPILE_OPTIONS}>
```

Comment:
This changes the current behavior. With the language option Kokkos will now be a `CXX` and a `CUDA/HIP` library. This also means that we add flags to a `CXX` target and a `CUDA/HIP` target. People might have been relying on this being only on `CUDA/HIP` tagets in their projects to separate code.

I will think about if we can make this composable.